src/acl/AnnotationData.cc

```diff
@@ -23,8 +24,8 @@ SBufList
 ACLAnnotationData::dump() const
 {
     SBufList sl;
-    if (const char *strNotes = notes->toString())
-        sl.push_back(SBuf(strNotes));
+    if (!notes->empty())
+        sl.push_back(notes->dump());
```

Comment:
I assume this is the only place where we call the new Notes::dump() method. Please convert that method to accept an std::ostream reference and use SBufStream here to convert dump() output into an SBufList. The current PR code essentially does the same, but uses SBufStream inside Notes::dump() which complicates that method reuse and slows down the ongoing transition from SBufList to streams.